ddtrace/internal/telemetry/telemetry_writer.py

```diff
@@ -0,0 +1,128 @@
+import http
+import logging
+import os
+import typing
+
+from ...utils.time import StopWatch
+from ..agent import get_connection
+from ..compat import get_connection_response
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from .data.integration import Integration
+from .data.payload import AppClosedPayload
+from .data.payload import AppHeartbeatPayload
+from .data.payload import AppIntegrationsChangedPayload
+from .data.payload import AppStartedPayload
+from .data.request_body import RequestBody
+
+
+DEFAULT_TELEMETRY_ENDPOINT = "https://instrumentation-telemetry-intake.datadoghq.com"
+DEFAULT_TELEMETRY_ENDPOINT_TEST = "https://all-http-intake.logs.datad0g.com/api/v2/apmtelemetry"
+
+log = get_logger(__name__)
+
+
+class TelemetryWriter(PeriodicService):
```

Comment:
We should not create an instance of this class + start the background thread if it is disabled.

If we run the background thread it'll still lock the GIL even though it isn't doing anything.